calibration_residual_kernals.py

```diff
@@ -83,10 +104,10 @@ def calc_residuals_from_ifft(ant_to_gains, observation_frequencies):
                 residual_delays[stream_idx] = -1.0 * tlags[max_idxs[stream_idx]]
                 gain_from_residual_delay = np.exp(2j*np.pi*(observation_frequencies[tune,:]*1e-9)*residual_delays[stream_idx])
                 zero_gains_indices = np.where(gain_matrix[stream_idx,:]==0.0)
-                residual_phases[stream_idx] = np.angle(gain_matrix[stream_idx,:]/gain_from_residual_delay)
-                residual_phases[stream_idx, zero_gains_indices] = 0.0
+                phase_cals[stream_idx] = np.angle(new_gain_matrix[stream_idx,:]/gain_from_residual_delay)
+                phase_cals[stream_idx, zero_gains_indices] = 0.0
```

Comment:
What is no longer working I believe is this line. You'll see that albeit that it is outside the observable field, the old method retains the original value, while the new method does not. It also zero's out some other areas where the gain matrix is zero... should perhaps instead zero out where `new_gain_matrix == 0.0`?

Reply:
I'm not sure either approach is really ideal.  Maybe better would be something like interpolate phase vs freq if there is a zeroed channel range with non-zero values on either side.  For the regions on the edge of the band where there is zero data I think we just leave it at zero.  It doesn't matter whether you do the test on `gain_matrix` or `new_gain_matrix` the amplitudes should be the same.

Reply:
Yeah, I'd say best to worst: interpolate, use old coeffs, use zero coeffs.